### PR TITLE
Models restructure as part of Configurations revamp - AnalysisModels.js

### PR DESCRIPTION
**Pull request by angelac2023**
Please let us know if there are any changes to be made.

### CODE REVIEW

File: leaf-ui/rappid-extensions/AnalysisModels.js [hunk 1]
```diff
@@ -13,9 +13,7 @@ var ResultModel = Backbone.Model.extend({
     idAttribute: "uid",
 
     defaults: {
-        name:"Default Result",
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/AnalysisModels.js:16]: Probably still needs the name and selected params for now to allow you to switch between results and have a name that renders on the sidebar

File: leaf-ui/rappid-extensions/AnalysisModels.js [hunk 2]
```diff
@@ -92,6 +85,25 @@ var ConfigModel = Backbone.Model.extend({
     },
 });
 
+/**
+ * Analysis Parameters 
+ */
+
+var ConfigAnalysis = Backbone.Model.extend ({ 
+    initialize : function(options){
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/AnalysisModels.js]: I think we might need the action param in here too @amgrubb can you confirm?

File: leaf-ui/rappid-extensions/AnalysisModels.js [hunk 3]
```diff
@@ -92,6 +85,25 @@ var ConfigModel = Backbone.Model.extend({
     },
 });
 
+/**
+ * Analysis Parameters 
+ */
+
+var ConfigAnalysis = Backbone.Model.extend ({
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/AnalysisModels.js]:
```suggestion
var AnalysisParametersBBM = Backbone.Model.extend ({ 
```

File: leaf-ui/rappid-extensions/AnalysisModels.js [hunk 4]
```diff
@@ -92,6 +85,25 @@ var ConfigModel = Backbone.Model.extend({
...
+/**
+ * Analysis Parameters 
+ */
+
+var ConfigAnalysis = Backbone.Model.extend ({ 
+    initialize : function(options){
+        _.extend({}, this.defaults, options); 
+    },
+
+    idAttribute: "uid",
```
Comment (leshghi) [inline leaf-ui/rappid-extensions/AnalysisModels.js]:
```suggestion
```

File: leaf-ui/rappid-extensions/AnalysisModels.js (hunk 1 above)
Comment (leshghi) [reply leaf-ui/rappid-extensions/AnalysisModels.js:16]: I also think its possible we may want the Result to hold the AnalysisParametersBBM. @amgrubb thoughts?

**Review by leshghi — CHANGES_REQUESTED** [review]
Some naming convention changes and moving things around. I only looked at the model restructure file because that is what we are reviewing. The buttons seem to be in progress and I think can wait/be separated from this PR.

File: leaf-analysis/app.js [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 // Name of .jar file for BloomingLeaf project must be Blooming.jar
 //var userPath = "/Users/<your user path here>/BloomingLeaf"
-var userPath = "/Users/judySmith/git/BloomingLeaf"
+var userPath = "/Users/ChuAn_nu0wj7q/Desktop/BloomingLeaf"
```
Comment (amgrubb) [inline leaf-analysis/app.js:7]: Please remove this change.

File: leaf-ui/index.html [hunk 6]
```diff
@@ -170,13 +184,16 @@
         <script src="./rappid-extensions/BackboneModelsOther.js"></script>
         <script src="./rappid-extensions/AnalysisModels.js"></script>
         <script src="./rappid-extensions/AssignmentsTable.js"></script>
+        <!-- new addition -->
```
Comment (amgrubb) [inline leaf-ui/index.html]: You can remove this line.

File: leaf-ui/rappid-extensions/AnalysisModels.js (hunk 1 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/AnalysisModels.js:16]: Yes, we need a default name, for now.
Also, we discussed having default analysis parameters and then when the analysis is called, saving the results with the analysis parameters.

**Review by amgrubb — COMMENTED** [review]
I'll take another look another day.

File: leaf-ui/rappid-extensions/AnalysisModels.js (hunk 2 above)
Comment (amgrubb) [reply leaf-ui/rappid-extensions/AnalysisModels.js]: @leshghi you mean to distinguish between single path and next states? Yes, we should add the action parameter.

Comment (amgrubb) [reply leaf-ui/rappid-extensions/AnalysisModels.js]: Good catch.

**Comment by amgrubb** [comment]
Replaced by #461

**Pull request closed without merging**